src/components/Select2.js

```diff
@@ -39,13 +44,24 @@ export default class Select2 extends Component {
     this.el = null;
   }
 
-  componentDidMount() {
+  componentDidMount() {    
     this.el = $(ReactDOM.findDOMNode(this));
     this.el.select2(this.props.options);
 
     this.props.events.forEach(event => {
       this.el.on(event[0], this.props[event[1]]);
     });
+    if (this.el.val() !== this.props.value) {
+      this.el.val(this.props.value).trigger("change");
```

Comment:
In this case, if you specify `defaultValue`, it will not be set.
Need to check a value is specified.